src/s390x/dasd.rs

```diff
@@ -33,18 +32,30 @@ use crate::runcmd;
 // IBM DASD Support
 /////////////////////////////////////////////////////////////////////////////
 
+const ECKD_DASD_BLOCKSIZE: u32 = 4096;
+
 #[derive(Debug)]
 struct Range {
     in_offset: u64,
     out_offset: u64,
     length: u64,
 }
 
-pub fn prepare_dasd(config: &InstallConfig) -> Result<()> {
-    low_level_format(&config.device)?;
-    if is_invalid(&config.device)? {
-        eprintln!("Disk {} is invalid, formatting", &config.device);
-        default_format(&config.device)?
+/// Returns default block size ony for unformatted ECKD DASD disk,
+/// otherwise None
+pub fn unformatted_sector_size(device: &str) -> Result<Option<NonZeroU32>> {
```

Comment:
The function name implies that the caller should already know whether the device is formatted, but that's this function's job.  What about something like `dasd_try_get_sector_size()`?